pkg/storage/config/schema_config.go

```diff
@@ -240,17 +240,32 @@ func usingForPeriodConfigs(configs []PeriodConfig, fn func(PeriodConfig) bool) b
 	return false
 }
 
+func usingObjectStoreIndex(cfg PeriodConfig) bool {
+	switch cfg.IndexType {
+	case BoltDBShipperType, TSDBType:
+		return true
+	default:
+		return false
+	}
+}
+
+// UsingObjectStorageIndex returns true if the current or upcoming period
+// uses object store index.
 func UsingObjectStorageIndex(configs []PeriodConfig) bool {
-	fn := func(cfg PeriodConfig) bool {
-		switch cfg.IndexType {
-		case BoltDBShipperType, TSDBType:
-			return true
-		default:
-			return false
+	return usingForPeriodConfigs(configs, usingObjectStoreIndex)
+}
+
+// FilterObjectStoreIndex returns a new slice retaining only the periodConfigs
+// that use object store index.
+func FilterObjectStoreIndex(in []PeriodConfig) []PeriodConfig {
+	var out []PeriodConfig
```

Comment:
```suggestion
	out := make([]PeriodConfig, 0, len(in))
```

Let's preallocate where possible to avoid unnecessary slice growth